### PR TITLE
feat(windows): kmconfig GUI (in kmshell)

### PR DESCRIPTION
**Pull request by mcdurdin**
This adds a GUI for controlling all Keyman Settings, accessible through Keyman Configuration.

A new button appears `Keyman System Settings...`: 

image

After clicking, KMShell will elevate and present:

image

I've disabled Development Settings in this screenshot. This is tweak mode stuff. The description tells the user what is valid, but values are not checked (except for type -- no strings in an int field...)

Of particular interest is the engine.compatibility.text_services_framework entry. Selecting this exposes an additional button `Add TSF Override...`:

image

Clicking that button opens the following dialog:

image

The user can enter a filename such as "chrome.exe", and select a Value:

image

When they click OK, they'll see a new entry for Chrome:

image

Clicking `<Apply>` saves changes to the registry.

The user can reset any or all settings, as well as exporting or importing to/from file. 

Show Development Settings throws a bunch of additional settings up which are only useful for us:

image

### CODE REVIEW

**Comment by darcywong00** [comment]
> The user can reset any or all settings, as well as exporting or importing to/from file.

What about changing the <kbd>Reset</kbd> label to <kbd>Reset to default</kbd> to clarify what it does?

File: windows/src/desktop/kmshell/settings/Keyman.Configuration.UI.UfrmSettingsManager.dfm [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    ScrollBars = ssVertical
+    TabOrder = 1
+  end
+  object cmdApply: TButton
+    Left = 528
+    Top = 414
+    Width = 87
+    Height = 25
+    Anchors = [akRight, akBottom]
+    Caption = '&Apply'
```
Comment (darcywong00) [inline windows/src/desktop/kmshell/settings/Keyman.Configuration.UI.UfrmSettingsManager.dfm:139]: Is there a TODO to have the captions in strings.xml?

**Review by darcywong00 — APPROVED** [review]
lgtm

File: windows/src/desktop/kmshell/settings/Keyman.Configuration.UI.UfrmSettingsManager.dfm (hunk 1 above)
Comment (mcdurdin) [reply windows/src/desktop/kmshell/settings/Keyman.Configuration.UI.UfrmSettingsManager.dfm:139]: See #3758.

**Pull request merged**